VEDA_OS01/views.py

```diff
@@ -96,6 +109,107 @@ def perform_create(self, serializer):
+    """
+
+    def get_cielo_token_response(self, username, api_secure_key):
+        """
+        Returns Cielo24 api token.
+
+        Arguments:
+            username(str): Cielo24 username
+            api_securekey(str): Cielo24 api key
+
+        Returns:
+            Response : Http response object
+        """
+        return requests.get(CIELO24_LOGIN_URL, params={
+            'v': CIELO24_API_VERSION,
+            'username': username,
+            'securekey': api_secure_key
+        })
+
+    def validate_transcript_credentials(self, provider, **credentials):
+        """
+        Validates transcript credentials.
+
+        Validations:
+            Providers must be either 3PlayMedia or Cielo24.
+            In case of:
+                3PlayMedia - 'api_key' and 'api_secret_key' are required.
+                Cielo24 - 'api_key' and 'username' are required.
+        """
+        error_message, validated_credentials = '', {}
+        valid_providers = [TranscriptProvider.CIELO24, TranscriptProvider.THREE_PLAY]
+        if provider in valid_providers:
+            attrs = ('org', 'api_key')
+            attrs += ('username', ) if provider == TranscriptProvider.CIELO24 else ('api_secret_key', )
+            missing = [attr for attr in attrs if attr not in credentials]
+            if missing:
+                error_message = u'{missing} must be specified for {provider}.'.format(
+                    provider=provider,
+                    missing=' and '.join(missing)
+                )
+
+                return error_message, validated_credentials
+           
+            validated_credentials.update({
+                prop: credentials[prop] for prop in attrs
+            })
+        else:
+            error_message = u'Invalid provider {provider}.'.format(provider=provider)
+
+        return error_message, validated_credentials
+
+    def post(self, request):
+        """
+        Creates or updates the org-specific transcript credentials with the given information.
+
+        Arguments:
+            request: A WSGI request.
+        """
+        # Validate credentials
+        provider = request.data.pop('provider', None)
+        error_message, validated_credentials = self.validate_transcript_credentials(provider=provider, **request.data)
+        if error_message:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data=dict(message=error_message)
+            )
+
+        org = validated_credentials['org']
+        api_key = validated_credentials['api_key']
+        username = validated_credentials.get('username')
+        api_secret_key = validated_credentials.get('api_secret_key')
+
+        # If Cielo, get Cielo api token and store it in api_key.
+        if provider == TranscriptProvider.CIELO24:
+            response = self.get_cielo_token_response(username=username, api_secure_key=api_key)
```

Comment:
I think this should stay as a part of validation and we just pass validated data to credentials model.